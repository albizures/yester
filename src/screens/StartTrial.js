import React, { Component } from 'react'
import {
  ScrollView,
  StyleSheet,
  TouchableHighlight,
  Text, View, Dimensions,
} from 'react-native'
import PropTypes from 'prop-types'

import Translate from '../components/Translate'
import styles from '../styles/common'

class StartTrial extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
  }

  onPressSubsciption = () => {
    this.props.navigation.navigate('Subscription')
  }

  onPressLogin = () => {
    this.props.navigation.navigate('Login')
  }

  render () {
    return (
      <View style={localStyles.container2}>
        <ScrollView style={[localStyles.container]}>
          <Text style={[styles.display2, localStyles.margin]}>
            <Translate keyName='common.upperTitle' />
          </Text>
          <Text style={[styles.title, localStyles.margin]}>
            <Translate keyName='startTrial.weKnow' />
            <Text style={{fontWeight: 'bold'}}>
              <Translate keyName='startTrial.weWant' />
            </Text>
          </Text>
          <Text style={[styles.title, localStyles.margin]}>
            <Translate keyName='startTrial.receive' />
          </Text>
          <TouchableHighlight onPress={this.onPressSubsciption}
            style={[styles.button, styles.filledButton]}>
            <Text style={[styles.buttonText]}>
              <Translate keyName='startTrial.start' />
            </Text>
          </TouchableHighlight>

          <Text style={[styles.body1, localStyles.margin]}>
            <Translate keyName='startTrial.price' />
          </Text>

          <TouchableHighlight onPress={this.onPressLogin}
            style={[styles.button, styles.outlinedButton, localStyles.margin]}>
            <Text style={[styles.buttonText, styles.notColor]}>
              <Translate keyName='startTrial.logIn' />
            </Text>
          </TouchableHighlight>

          <View style={[styles.separator, localStyles.margin]} />
          <Text style={[styles.body2, {textAlign: 'left', marginBottom: height * 0.01}]}>
            <Translate keyName='startTrial.cancel' />
          </Text>
          <Text style={[styles.body1, {textAlign: 'left'}]}>
            <Translate keyName='startTrial.recurring' />
          </Text>
        </ScrollView>
        <View style={localStyles.container3}>
          <Text style={[styles.title, {textAlign: 'center'}]}>
            <Translate keyName='startTrial.already' />
            <Text style={[{fontWeight: 'bold', textDecorationLine: 'underline'}]}>
              <Translate keyName='startTrial.restore' />
            </Text>
          </Text>
        </View>
      </View>
    )
  }
}

let { height, width } = Dimensions.get('window')
const localStyles = StyleSheet.create({
  container: {
    height: height * 0.92,
    paddingTop: height * 0.075,
    paddingHorizontal: width * 0.08,
    flex: 1,
    flexDirection: 'column',
    zIndex: 0,
  },
  container2: {
    position: 'absolute',
    backgroundColor: '#fff',
  },
  container3: {
    position: 'absolute',
    marginTop: height * 0.92,
    height: height * 0.08,
    width: width,
    zIndex: 2,
    backgroundColor: '#BDE0FC',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  margin: {
    marginBottom: height * 0.03,
  },
})

export default StartTrial
