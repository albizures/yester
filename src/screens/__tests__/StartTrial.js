import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { TouchableHighlight } from 'react-native'

import StartTrial from '../StartTrial'
const navigation = {
  navigate: jest.fn(),
}

describe('src/screens/StartTrial.js', () => {
  it('should render', () => {
    const tree = renderer.create(<StartTrial navigation={navigation} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('when button to free trail is pressed', () => {
    it('should navigate to StartTrial screen', () => {
      const shallowLogin = shallow(<StartTrial navigation={navigation} />)
      const shallowTouchableHighlights = shallowLogin.find(TouchableHighlight)
      const shallowButtonToSubscription = shallowTouchableHighlights.at(0)
      const shallowButtonToLogin = shallowTouchableHighlights.at(1)

      shallowButtonToLogin.prop('onPress')()
      shallowButtonToSubscription.prop('onPress')()

      expect(navigation.navigate).toHaveBeenCalledTimes(2)
      expect(navigation.navigate).toHaveBeenCalledWith('Subscription')
      expect(navigation.navigate).toHaveBeenCalledWith('Login')
    })
  })
})
