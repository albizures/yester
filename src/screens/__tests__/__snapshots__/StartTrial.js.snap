// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/StartTrial.js should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "position": "absolute",
    }
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column",
          "height": 1227.28,
          "paddingHorizontal": 60,
          "paddingTop": 100.05,
          "zIndex": 0,
        },
      ]
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#309BF8",
              "fontSize": 36,
              "fontWeight": "bold",
              "textAlign": "left",
            },
            Object {
              "marginBottom": 40.019999999999996,
            },
          ]
        }
      >
        <Text
          keyName="caps_title1"
        >
          YESTER
        </Text>
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#309BF8",
              "fontSize": 20,
              "textAlign": "left",
            },
            Object {
              "marginBottom": 40.019999999999996,
            },
          ]
        }
      >
        <Text
          keyName="before_lbl1_1"
        >
          We know you have beautiful and interesting story. 
        </Text>
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          <Text
            keyName="before_lbl1_2"
          >
            We want to help you, writing yours.
          </Text>
        </Text>
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#309BF8",
              "fontSize": 20,
              "textAlign": "left",
            },
            Object {
              "marginBottom": 40.019999999999996,
            },
          ]
        }
      >
        <Text
          keyName="before_lbl2"
        >
          Receive one conversation starter a day. Try it for FREE a month (then just $4.99/month).
        </Text>
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "padding": 26.68,
            },
            Object {
              "backgroundColor": "#309BF8",
              "borderColor": "#309BF8",
              "borderRadius": 53.36,
              "borderWidth": 2,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          <Text
            keyName="before_lbl7"
          >
            START FREE TRIAL
          </Text>
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#309BF8",
              "fontSize": 16,
              "textAlign": "center",
            },
            Object {
              "marginBottom": 40.019999999999996,
            },
          ]
        }
      >
        <Text
          keyName="before_lbl3"
        >
          $4.99/month after trial ends.
        </Text>
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "padding": 26.68,
            },
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#309BF8",
              "borderRadius": 53.36,
              "borderWidth": 2,
            },
            Object {
              "marginBottom": 40.019999999999996,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "color": "#309BF8",
              },
            ]
          }
        >
          <Text
            keyName="before_lbl8"
          >
            Log In
          </Text>
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#309BF8",
              "borderBottomWidth": 1,
            },
            Object {
              "marginBottom": 40.019999999999996,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#309BF8",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {
              "marginBottom": 13.34,
              "textAlign": "left",
            },
          ]
        }
      >
        <Text
          keyName="before_lbl4"
        >
          Cancel Anytime
        </Text>
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#309BF8",
              "fontSize": 16,
              "textAlign": "center",
            },
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        <Text
          keyName="before_lbl5"
        >
          At Yester is recurring subscription. If you decide to purchase a subscription, an initial payment will be...
        </Text>
      </Text>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#BDE0FC",
        "flex": 1,
        "height": 106.72,
        "justifyContent": "center",
        "marginTop": 1227.28,
        "position": "absolute",
        "width": 750,
        "zIndex": 2,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#309BF8",
            "fontSize": 20,
            "textAlign": "left",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      <Text
        keyName="before_lbl6_1"
      >
        Already purchased? 
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        <Text
          keyName="before_lbl6_2"
        >
          Restore
        </Text>
      </Text>
    </Text>
  </View>
</View>
`;
